src/smt.rs

```diff
@@ -893,13 +905,9 @@ impl SparseMerkleTree {
         value_bin: &'a [&'a [u8]],
         height: Height,
     ) -> Result<Bins<'a>, SMTError> {
-        let mut keys = vec![];
-        let mut values = vec![];
+        let mut keys: Vec<Vec<&[u8]>> = vec![vec![]; self.max_number_of_nodes];
+        let mut values: Vec<Vec<&[u8]>> = vec![vec![]; self.max_number_of_nodes];
```

Comment:
We already have some alias type for nested vectors, so I think it would be better to have a new type for `Vec<Vec<&[u8]>>` or we can change `NestedVec` to accept generic time like `ArcMutex`

Reply:
Thank you! Done.